llvm/test/CodeGen/SBF/static_syscall_2.ll

```diff
@@ -0,0 +1,50 @@
+; RUN: llc -march=sbf -mattr=+static-syscalls < %s | FileCheck --check-prefix=CHECK %s
+
+; Syscall declaration in C:
+;
+; int c_declaration(int a, int b) {
+;    int (*const syscall)(int a, int b) = (void*)50;
+;    return syscall(a, b);
+; }
+; The following is the unoptimized output from clang:
+
+define dso_local i32 @c_declaration(i32 noundef %a, i32 noundef %b) #0 {
+entry:
+  %a.addr = alloca i32, align 4
+  %b.addr = alloca i32, align 4
+  %syscall = alloca ptr, align 8
+  store i32 %a, ptr %a.addr, align 4
+  store i32 %b, ptr %b.addr, align 4
+  store ptr inttoptr (i64 50 to ptr), ptr %syscall, align 8
+  %0 = load i32, ptr %a.addr, align 4
+  %1 = load i32, ptr %b.addr, align 4
+
+; Ensure the syscall instruction is emitted
+; CHECK: syscall 50
+
+  %call = call i32 inttoptr (i64 50 to ptr)(i32 noundef %0, i32 noundef %1)
+  ret i32 %call
+}
+
+; Syscall declaration in Rust:
+;
+;  #[no_mangle]
+;  pub unsafe fn rust_declaration(b: u64) -> u32 {
+;      let syscall : extern "C" fn(b: u64) -> u32 = core::mem::transmute(60u64);
+;      return syscall(b);
+;  }
+; The following is the unoptimized output from rustc:
```

Comment:
In some circumstances, specially when we build the code with `-O0`, the front end may emit a `callx`. Instead of creating a peephole or forcing optimizations in the backend, I decided to add the correct declaration example in both C and Rust.

These examples generate LLVM-IR code that even when unoptimized will create a `syscall` instruction.